### ISSUE TITLE
Problems with library built-in types (TS1005)

### ISSUE THREAD

**Issue by pedromoraisf** [Unleash/proxy-client-react#73]
### Describe the bug

After installing the library, when using the `useUnleashContext` function, I am getting an error related with internal types of this library .

### Steps to reproduce the bug

1. Run `npm i --save unleash-proxy-client @unleash/proxy-client-react`
2. Import `useUnleashContext` in any file

### Logs, error output, etc.

```bash
TypeScript error in /Users/pedromoraisf/www/work/admission/gupy-front-admission/admission-companies/node_modules/@unleash/proxy-client-react/dist/src/FlagContext.d.ts(2,13):
'=' expected. TS1005

     1 | import React from 'react';
   > 2 | import type { UnleashClient } from 'unleash-proxy-client';
       | ^
     3 | export interface IFlagContextValue extends Pick<UnleashClient, 'on' | 'updateContext' | 'isEnabled' | 'getVariant'> {
     4 | client: UnleashClient;
     5 | flagsReady: boolean;
```

### Additional context

- I'm using `skipLibCheck: true` in `tsconfig.json`
- `node_modules` has in `exclude` option, on `tsconfig.json`
- Searching on the web, i have found [this page](https://github.com/microsoft/TypeScript/issues/41883), where one person thinks the problem has to do with some type declaration in the `package.json` of that lib (`@unleash/proxy-client-react`) 

### Unleash version

4.10.5

**Comment by sjaanus** [comment]
Hey @pedromoraisf . I created an example project based on your setup, but could not reproduce this TS error. Here it is https://github.com/sjaanus/proxy-client-react-73. Does this repository also throw you the same error? Hopefully we will catch the cause of your issue. :smiley:

**Comment by pedromoraisf** [comment]
Hey @sjaanus, how are you?

I managed to resolve the errors. For the error in the issue description, I got the resolution by upgrading to TypeScript `3.9.10` (previously it was at version `3.7.4`).

Then the following error started happening:
![Screenshot from 2022-08-16 15-45-28](https://user-images.githubusercontent.com/55749387/185395783-0784ed3b-27c1-4f9b-b6a0-749392de1d37.png)

The solution was to encapsulate `UnleashProvider` very high in the React render tree hierarchy. I started right after the component that starts `redux`, in the application.
